test/object-store/util/sync/baas_admin_api.cpp

```diff
@@ -341,6 +389,203 @@ app::Response do_http_request(const app::Request& request)
+    Baasaas& operator=(Baasaas&&) = delete;
+
+    ~Baasaas()
+    {
+        stop();
+    }
+
+    void poll()
+    {
+        if (!m_http_endpoint.empty() || m_container_id.empty()) {
+            return;
+        }
+
+        auto logger = util::Logger::get_default_logger();
+        auto poll_start_at = std::chrono::system_clock::now();
+        std::string http_endpoint;
+        std::string mongo_endpoint;
+        bool logged = false;
+        while (std::chrono::system_clock::now() - poll_start_at < std::chrono::minutes(2) &&
+               m_http_endpoint.empty()) {
+            if (http_endpoint.empty()) {
+                auto status_obj =
+                    do_request(util::format("containerStatus?id=%1", m_container_id), app::HttpMethod::get);
+                if (!status_obj["httpUrl"].is_null()) {
+                    http_endpoint = status_obj["httpUrl"].get<std::string>();
+                    mongo_endpoint = status_obj["mongoUrl"].get<std::string>();
+                }
+            }
+            else {
+                app::Request baas_req;
+                baas_req.url = util::format("%1/api/private/v1.0/version", http_endpoint);
+                baas_req.method = app::HttpMethod::get;
+                baas_req.headers.insert_or_assign("Content-Type", "application/json");
+                auto baas_resp = do_http_request(baas_req);
+                if (baas_resp.http_status_code >= 200 && baas_resp.http_status_code < 300) {
+                    m_http_endpoint = http_endpoint;
+                    m_mongo_endpoint = mongo_endpoint;
+                    break;
+                }
+            }
+
+            if (!logged) {
+                logger->info("Waiting for baasaas container \"%1\" to be ready", m_container_id);
+                logged = true;
+            }
+            std::this_thread::sleep_for(std::chrono::seconds(3));
+        }
+
+        if (m_http_endpoint.empty()) {
+            throw std::runtime_error(
+                util::format("Failed to launch baasaas container %1 within 2 minutes", m_container_id));
+        }
+    }
+
+    void stop()
+    {
+        auto container_id = std::move(m_container_id);
+        if (container_id.empty()) {
+            return;
+        }
+
+        auto logger = util::Logger::get_default_logger();
+        logger->info("Stopping baasaas container with id \"%1\"", container_id);
+        do_request(util::format("stopContainer?id=%1", container_id), app::HttpMethod::post);
+    }
+
+    const std::string& http_endpoint()
+    {
+        poll();
+        return m_http_endpoint;
+    }
+
+    const std::string& mongo_endpoint()
+    {
+        poll();
+        return m_mongo_endpoint;
+    }
+
+private:
+    nlohmann::json do_request(std::string api_path, app::HttpMethod method)
+    {
+        app::Request request;
+        request.url = util::format(
+            "https://us-east-1.aws.data.mongodb-api.com/app/baas-container-service-autzb/endpoint/%1", api_path);
```

Comment:
can you leave a comment about who owns this app and if there is any documentation about the endpoints?

Reply:
done. I've also made it so we can override this via an env variable.